packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -1182,151 +1183,305 @@ export class WorkflowExecute {
-		} else if (nodeType.trigger) {
-			if (mode === 'manual') {
-				// In manual mode start the trigger
-				const triggerResponse = await Container.get(TriggersAndPollers).runTrigger(
-					workflow,
-					node,
-					NodeExecuteFunctions.getExecuteTriggerFunctions,
-					additionalData,
-					mode,
-					'manual',
-				);
+			return { data: response, closeFunction };
+		}
+		// For trigger nodes in any mode except "manual" do we simply pass the data through
+		return { data: inputData.main as INodeExecutionData[][] };
+	}
 
-				if (triggerResponse === undefined) {
-					return { data: null };
-				}
+	/**
+	 * Executes a declarative node using RoutingNode
+	 */
+	private async executeDeclarativeNode(
+		workflow: Workflow,
+		node: INode,
+		nodeType: INodeType,
+		additionalData: IWorkflowExecuteAdditionalData,
+		mode: WorkflowExecuteMode,
+		runExecutionData: IRunExecutionData,
+		runIndex: number,
+		connectionInputData: INodeExecutionData[],
+		inputData: ITaskDataConnections,
+		executionData: IExecuteData,
+	): Promise<IRunNodeResponse> {
+		// NOTE: This block is only called by nodes tests.
+		// In the application, declarative nodes get assigned a `.execute` method in NodeTypes.
+		const context = new ExecuteContext(
+			workflow,
+			node,
+			additionalData,
+			mode,
+			runExecutionData,
+			runIndex,
+			connectionInputData,
+			inputData,
+			executionData,
+			[],
+		);
+		const routingNode = new RoutingNode(context, nodeType);
+		const data = await routingNode.runNode();
+		return { data };
+	}
 
-				let closeFunction;
-				if (triggerResponse.closeFunction) {
-					// In manual mode we return the trigger closeFunction. That allows it to be called directly
-					// but we do not have to wait for it to finish. That is important for things like queue-nodes.
-					// There the full close will may be delayed till a message gets acknowledged after the execution.
-					// If we would not be able to wait for it to close would it cause problems with "own" mode as the
-					// process would be killed directly after it and so the acknowledge would not have been finished yet.
-					closeFunction = triggerResponse.closeFunction;
-
-					// Manual testing of Trigger nodes creates an execution. If the execution is cancelled, `closeFunction` should be called to cleanup any open connections/consumers
-					abortSignal?.addEventListener('abort', closeFunction);
-				}
+	/**
+	 * Executes the given node based on its type and configuration
+	 *
+	 * @param workflow - The workflow containing the node
+	 * @param executionData - Data needed for node execution
+	 * @param runExecutionData - Overall execution state
+	 * @param runIndex - Index of current execution run
+	 * @param additionalData - Additional workflow execution data
+	 * @param mode - Workflow execution mode
+	 * @param abortSignal - Signal for canceling execution
+	 * @returns Promise resolving to node execution result
+	 */
+	async runNode(
```

Comment:
We have a `runNode` (which orchestrates) and an `executeNode` (which actually executes). I find the above docline mostly redundant of the code (does it add much?), but it'd be a good opportunity to document this so folks can clearly differentiate between them. Or maybe finding better names that make this clear outright.